clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -86,9 +102,7 @@ PVConstraint *ConstraintResolver::addAtom(PVConstraint *PVC, Atom *PtrTyp, Const
       auto *Prem = CS.createGeq(NewA, CS.getWild());
       auto *Conc = CS.createGeq(VA, CS.getWild());
       CS.addConstraint(CS.createImplies(Prem, Conc));
-    } else if (ConstAtom *C = dyn_cast<WildAtom>(A)) {
-      NewA = CS.getWild();
-    } // else stick with what's given
+    }
```

Comment:
Can you explain why this code can be deleted? What problem does it solve?

Reply:
That line is a little puzzling. I added an `assert(false)` there to see what test even hit that branch, and it's only the tests I wrote dealing with `InitListExpressions`. The reason I removed it is because it caused `int x[1] = {(int*) 1}` to rewrite to itself. `(int*)1` is `WildAtom`, so the result of `addAtom` was forced to be the same. This was equated to `x`, even though `x` should not be wild.